tests/e2e/pkg/tests/mocks.go

```diff
@@ -0,0 +1,264 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+
+	//. "github.com/onsi/gomega"
+	"time"
+)
+
+// MockTestsMinimal:
+// 3 different provider 2 different version 3 different K8s cluster with 2 tests
+// Expected: 2 testCase 1 testCase on any available K8sCluster
+// 1x2 + 1x3 = 5 test all-together
+// Runtime parallel: 1x(5) = 5sec (time of the longest testCase)
+func MockTestsMinimal() Classifier {
+	k8sClusterPool := K8sClusterPool{}
+	k8sClusterPool.AddK8sClusters(
+		NewMockK8sCluster(
+			"testContextPath1",
+			"testContextName1",
+			"1.24",
+			"provider1",
+			"clusterID1",
+		),
+		NewMockK8sCluster(
+			"testContextPath2",
+			"testContextName2",
+			"1.25",
+			"provider2",
+			"clusterID2",
+		),
+		NewMockK8sCluster(
+			"testContextPath3",
+			"testContextName3",
+			"1.24",
+			"provider3",
+			"clusterID3",
+		),
+	)
+	return NewClassifier(k8sClusterPool, mockTest1, mockTest2)
+}
+
+// MockTestsProvider:
+// 3 different provider 3 different K8s cluster with 2 tests
```

Comment:
Same as above, it'd be great if we can rephrase this to make it more understandable

Reply:
fixed: [2b85e9f](https://github.com/banzaicloud/koperator/pull/1020/commits/2b85e9f4affdc52e1b2980983dcb990fa193bf2f)